components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/handler/TenantRegistrationVerificationHandler.java

```diff
@@ -0,0 +1,119 @@
+import org.wso2.carbon.identity.recovery.RecoveryScenarios;
+import org.wso2.carbon.identity.recovery.RecoverySteps;
+import org.wso2.carbon.identity.recovery.util.Utils;
+import org.wso2.carbon.registry.core.utils.UUIDGenerator;
+import org.wso2.carbon.user.core.UserStoreManager;
+
+import java.util.Map;
+
+/**
+ * This handler is used to send password reset link for tenant admin during tenant registration.
+ */
+public class TenantRegistrationVerificationHandler extends UserEmailVerificationHandler {
+
+    private static final Log log = LogFactory.getLog(TenantRegistrationVerificationHandler.class);
+
+    public String getName() {
+
+        return "tenantRegistrationVerificationHandler";
+    }
+
+    public String getFriendlyName() {
+
+        return "Tenant registration Email Verification";
+    }
+
+    @Override
+    public void handleEvent(Event event) throws IdentityEventException {
+
+        String eventName = event.getEventName();
+
+        if (log.isDebugEnabled()) {
+            log.debug("Handling event : " + eventName);
+        }
+
+        Map<String, Object> eventProperties = event.getEventProperties();
+        UserStoreManager userStoreManager = (UserStoreManager) eventProperties.get(IdentityEventConstants
+                .EventProperty.USER_STORE_MANAGER);
+
+        Map<String, String> claims = (Map<String, String>) eventProperties.get(IdentityEventConstants.EventProperty
+                .USER_CLAIMS);
+        if (IdentityEventConstants.Event.PRE_SET_USER_CLAIMS.equals(eventName) && Boolean.valueOf(claims.
+                get(IdentityRecoveryConstants.TENANT_ADMIN_ASK_PASSWORD_CLAIM))) {
+            handleClaimUpdate(eventProperties, userStoreManager, claims);
+        }
+    }
+
+    @Override
+    public void init(InitConfig configuration) throws IdentityRuntimeException {
+
+        super.init(configuration);
+    }
+
+    @Override
+    public int getPriority(MessageContext messageContext) {
+
+        return 26;
+    }
+
+    protected void handleClaimUpdate(Map<String, Object> eventProperties, UserStoreManager userStoreManager,
+                                     Map<String, String> claims) throws IdentityEventException {
+
+        User user = getUser(eventProperties, userStoreManager);
+
+        if (log.isDebugEnabled()) {
+            log.debug("PreSetUserClaim - TenantRegistrationVerificationHandler for : " + user.toString() + ". This is " +
+                    "an update request for the claim:  " + IdentityRecoveryConstants.TENANT_ADMIN_ASK_PASSWORD_CLAIM);
+        }
+
+        // Remove claim to prevent persisting this temporary claim.
+        claims.remove(IdentityRecoveryConstants.TENANT_ADMIN_ASK_PASSWORD_CLAIM);
+
+        String uuid = UUIDGenerator.generateUUID();
+        String notificationType;
+        Enum recoveryScenario = RecoveryScenarios.TENANT_ADMIN_ASK_PASSWORD;
+
+        notificationType = IdentityRecoveryConstants.NOTIFICATION_TYPE_TENANT_REGISTRATION_EMAIL_CONFIRM;
```

Comment:
define inline in line 101

Reply:
Do you mean notification type as invite-via-email?